The piper extension requires libonnxruntime, which is not available on
noble. Don't build the extension, don't test it.

Note: Patch is generated with "quilt refresh -U2"

--- a/setup/extensions.json
+++ b/setup/extensions.json
@@ -136,12 +136,4 @@
     },
     {
-        "name": "piper",
-        "sources": "calibre/utils/tts/piper.cpp",
-		"needs_c++": "17",
-        "libraries": "!piper_libs",
-        "lib_dirs": "!piper_lib_dirs",
-        "inc_dirs": "!piper_inc_dirs"
-    },
-    {
         "name": "html_as_json",
         "sources": "calibre/srv/html_as_json.cpp",
--- a/src/calibre/test_build.py
+++ b/src/calibre/test_build.py
@@ -128,8 +128,4 @@ class BuildTest(unittest.TestCase):
         del SSIPClient
 
-    def test_piper(self):
-        from calibre.utils.tts.piper import simple_test
-        simple_test()
-
     def test_zeroconf(self):
         import ifaddr
@@ -143,4 +139,5 @@ class BuildTest(unittest.TestCase):
             # libusb fails to initialize in containers without USB subsystems
             exclusions.update(set('libusb libmtp'.split()))
+        exclusions.add('piper')  # XtraDeb
         from importlib import import_module
         from importlib.resources import files
--- a/src/calibre/utils/run_tests.py
+++ b/src/calibre/utils/run_tests.py
@@ -195,4 +195,6 @@ class TestImports(unittest.TestCase):
         if not isbsd:
             exclude_modules.add('calibre.devices.usbms.hal')
+        exclude_modules.add('calibre.gui2.tts.piper')   # XtraDeb
+        exclude_modules.add('calibre.utils.tts.piper')  # XtraDeb
         d = os.path.dirname
         SRC = d(d(d(os.path.abspath(__file__))))
